src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/BufferedHtmlContent.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNet.Html.Abstractions;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.AspNet.Mvc.Rendering
+{
+    /// <summary>
+    /// Enumerable object collection which knows how to write itself.
+    /// </summary>
+    public class BufferedHtmlContent : IHtmlContent, IEnumerable<object>
+    {
+        private const int MaxCharToStringLength = 1024;
+        // This is not List<IHtmlContent> because that would lead to boxing all strings to IHtmlContent
+        // which is not space performant.
+        // internal for testing.
+        internal List<object> Entries { get; set; } = new List<object>();
+
+        /// <summary>
+        /// Appends the string to the collection.
+        /// </summary>
+        /// <param name="value">The <c>string</c> to be appended.</param>
+        public void Append([NotNull] string value)
+        {
+            Entries.Add(value);
+        }
+
+        /// <summary>
+        /// Appends a character array to the collection.
+        /// </summary>
+        /// <param name="value">The character array to be appended.</param>
+        /// <param name="index">The index from which the character array must be read.</param>
+        /// <param name="count">The count till which the character array must be read.</param>
+        /// <remarks>
+        /// Splits the character array into strings of 1KB length and appends them.
```

Comment:
Should we be specifying 1KB here when it's actually MaxCharToStringLength. Seems like an implementation detail that is being leaked.